packages/fxa-auth-server/lib/routes/utils/signin.js

```diff
@@ -496,29 +501,49 @@ module.exports = (
         const secret = accountRecord.primaryEmail.emailCode;
         const code = otpUtils.generateOtpCode(secret, otpOptions);
         const { timeZone } = request.app.geo;
+        const emailContext = {
+          acceptLanguage: request.app.acceptLanguage,
+          code,
+          deviceId,
+          flowId,
+          flowBeginTime,
+          redirectTo,
+          resume,
+          service,
+          timeZone,
+          uaBrowser: request.app.ua.browser,
+          uaBrowserVersion: request.app.ua.browserVersion,
+          uaOS: request.app.ua.os,
+          uaOSVersion: request.app.ua.osVersion,
+          uaDeviceType: request.app.ua.deviceType,
+          uid: sessionToken.uid,
+        };
+        const rpCmsConfig = await fetchRpCmsData(request, cmsManager, log);
 
         try {
-          await mailer.sendVerifyLoginCodeEmail(
-            accountRecord.emails,
-            accountRecord,
-            {
-              acceptLanguage: request.app.acceptLanguage,
-              code,
-              deviceId,
-              flowId,
-              flowBeginTime,
-              redirectTo,
-              resume,
-              service,
-              timeZone,
-              uaBrowser: request.app.ua.browser,
-              uaBrowserVersion: request.app.ua.browserVersion,
-              uaOS: request.app.ua.os,
-              uaOSVersion: request.app.ua.osVersion,
-              uaDeviceType: request.app.ua.deviceType,
-              uid: sessionToken.uid,
-            }
-          );
+          if (!rpCmsConfig || !rpCmsConfig.VerifyLoginCodeEmail) {
```

Comment:
Validation is done in Strapi that these values won't be undefined, but I wonder if we should have some additional checks just incase that requirement changes. My concern is that we might send an email with some blank values and it could look suspicious.